File: iosApp/kosmos/kosmos/UseCases/Implementation/GetApodImpl.swift
//
//  GetApodImpl.swift
//  kosmos
//
//  Created by Robin Caroff on 13/08/2019.
//  Copyright © 2019 Maxime Lumeau. All rights reserved.
//

import Foundation
import kore

class GetApodImpl: GetApod {
    
    private var job: Kotlinx_coroutines_core_nativeJob?
    private let nasaAPI : NasaApi = NasaAPIRemote()
    
    func getApod(completion: @escaping (APOD)->(), error: (()->())?) {
        job = nasaAPI.getAPOD{ apod in
            completion(apod)
            
            return .init()
        }
    }
    
    //TODO Handle cancellation
}

File: androidApp/src/main/java/xyz/mlumeau/kosmos/usecases/implementations/GetAPODImpl.kt
package xyz.mlumeau.kosmos.usecases.implementations

import xyz.mlumeau.kosmos.kore.NasaAPIRemote
import xyz.mlumeau.kosmos.kore.service.nasa.NasaApi
import xyz.mlumeau.kosmos.usecases.GetAPOD

class GetAPODImpl(
    private val nasaAPI: NasaApi = NasaAPIRemote()
) : GetAPOD {
    override suspend fun invoke() = nasaAPI.getAPOD()
}